pywr/parameters/_control_curves.pyx

```diff
@@ -548,3 +549,78 @@ cdef class ControlCurveParameter(BaseControlCurveParameter):
         return self._upper_bounds
 
 ControlCurveParameter.register()
+
+
+cdef class WeightedAverageProfileParameter(Parameter):
+    """Generates a weighted average daily profile from pairs of storage nodes and
+    profile parameters.
+
+    The contribution on each profile is weighted using the volume of the paired
```

Comment:
Should mention that this parameter is efficient by calculating the profile during `.reset()`, but this limits the types of parameters that can be used.